beacon-chain/sync/data_columns_sampling.go

```diff
@@ -49,6 +247,15 @@ func sortedListFromMap(m map[uint64]bool) []uint64 {
 	return result
 }
 
+func randomPeerFromMap(m map[peer.ID]bool) peer.ID {
```

Comment:
Instead of creating a whole slice from the map, then selecting a random item from the slice, why not:
- Selecting in advance a random number `r` in `[0, len(m)[`, then
- Iterating thru the map, and retrieve the `r`th element?